packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,187 @@
+  ceilMultiplyBy,
+  floorDivideBy,
+  floorMultiplyBy,
+  makeRatio,
+} from '@agoric/zoe/src/contractSupport/index.js';
+import { Far } from '@endo/marshal';
+// import { CONTRACT_ELECTORATE } from '@agoric/governance';
+
+import { AmountMath } from '@agoric/ertp';
+import { makeMakeCollectFeesInvitation } from '../collectFees.js';
+
+const { details: X } = assert;
+
+/**
+ * @file The Parity Stability Module supports efficiently minting/burning a
+ * stable token at a specified fixed ratio to a reference stable token, which
+ * thereby acts as an anchor to provide additional stability. For flexible
+ * economic policies, the fee percentage for trading into and out of the stable
+ * token are specified separately.
+ *
+ */
+
+const BASIS_POINTS = 10000n;
+
+/**
+ * Stage a transfer of a singe asset from one seat to another, with an optional
+ * remapping of the Keywords. Check that the remapping is for the same amount.
+ *
+ * @param {ZCFSeat} from
+ * @param {ZCFSeat} to
+ * @param {AmountKeywordRecord} txFrom
+ * @param {AmountKeywordRecord} txTo
+ */
+function stageTransfer(from, to, txFrom, txTo = txFrom) {
+  assert(AmountMath.isEqual(Object.values(txFrom)[0], Object.values(txTo)[0]));
+  from.decrementBy(txFrom);
+  to.incrementBy(txTo);
+}
+
+/**
+ * @param {ZCF<{
+ *    anchorBrand: Brand,
+ *    anchorPerStable: Ratio,
+ *    main: {
+ *      WantStableFeeBP: bigint,
+ *      GiveStableFeeBP: bigint,
+ *      MintLimit: Amount } }>
+ * } zcf
+ * @param {{feeMintAccess: FeeMintAccess}} privateArgs
+ */
+export const start = async (zcf, privateArgs) => {
+  const { anchorBrand, anchorPerStable } = zcf.getTerms();
+
+  const { feeMintAccess } = privateArgs;
+  // TODO should this know that the name is 'Stable'
+  // TODO get the RUN magic out of here so the contract is more reusable
+  const stableMint = await zcf.registerFeeMint('Stable', feeMintAccess);
+  const { brand: stableBrand } = stableMint.getIssuerRecord();
+  zcf.setTestJig(() => ({
+    stableIssuerRecord: stableMint.getIssuerRecord(),
+  }));
+  const emptyStable = AmountMath.makeEmpty(stableBrand);
+  const emptyAnchor = AmountMath.makeEmpty(anchorBrand);
+
+  // HACK for simple governance API
+  const getGovernance = _ => {
+    const {
+      main: { WantStableFeeBP, GiveStableFeeBP, MintLimit },
+    } = zcf.getTerms();
+    assert(
+      AmountMath.isGTE(MintLimit, emptyAnchor),
+      X`MintLimit ${MintLimit} must specificy a limit in ${anchorBrand}`,
+    );
+    return {
+      getWantStableRate: () =>
+        makeRatio(WantStableFeeBP, stableBrand, BASIS_POINTS),
+      getGiveStableRate: () =>
+        makeRatio(GiveStableFeeBP, stableBrand, BASIS_POINTS),
+      getMintLimit: () => MintLimit,
+    };
+  };
+  const gov = getGovernance(zcf);
+
+  const { zcfSeat: anchorPool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: feePool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: stage } = zcf.makeEmptySeatKit();
+
+  const assertUnderLimit = anchorAmount => {
```

Comment:
Is this intended to prevent minting that adds up to more than mint limit, or just individual transactions that are larger than the limit? It seems to be doing the latter, and I'm not sure why that's useful. As a client, I can make multiple requests to stay under the limit.

Reply:
It prevents minting a total that gets to be more than the mint limit. It adds the incoming anchor amount to the total that's currently on the anchor pool, and asserts if that is over the limit. At least it's supposed to, which matches the first criteria.  I moved the addition into `assertUnderLimit` so it should be more clear now.